modules/setting/markup.go

```diff
@@ -25,42 +28,113 @@ type MarkupParser struct {
 	IsInputFile    bool
 }
 
+// MarkupSanitizer defines the policy for whitelisting attributes on
+// certain elements.
+type MarkupSanitizer struct {
+	Elements   []string
+	AllowAttrs []string
+	Regexps    []*regexp.Regexp
+}
+
 func newMarkup() {
-	extensionReg := regexp.MustCompile(`\.\w`)
 	for _, sec := range Cfg.Section("markup").ChildSections() {
 		name := strings.TrimPrefix(sec.Name(), "markup.")
 		if name == "" {
 			log.Warn("name is empty, markup " + sec.Name() + "ignored")
 			continue
 		}
 
-		extensions := sec.Key("FILE_EXTENSIONS").Strings(",")
-		var exts = make([]string, 0, len(extensions))
-		for _, extension := range extensions {
-			if !extensionReg.MatchString(extension) {
-				log.Warn(sec.Name() + " file extension " + extension + " is invalid. Extension ignored")
-			} else {
-				exts = append(exts, extension)
-			}
+		if name == "sanitizer" {
+			newMarkupSanitizer(name, sec)
+		} else {
+			newMarkupRenderer(name, sec)
 		}
+	}
+}
+
+func newMarkupSanitizer(name string, sec *ini.Section) {
+	// Implementation detail: go-ini/ini doesn't currently support
+	// ini arrays properly: https://github.com/go-ini/ini/issues/197
```

Comment:
This comment is confusing because it actually doed support them as "shadows" and you're using them below.